tensornetwork/network_components.py

```diff
@@ -256,9 +257,20 @@ def has_nondangling_edge(self):
         return True
     return False
 
+  @overload
+  def __getitem__(self, key: slice) -> List["Edge"]:
+    return self.edges[key]
+  
+  @overload
   def __getitem__(self, key: Union[int, Text]) -> "Edge":
     return self.get_edge(key)
 
+  def __getitem__(self, key: Union[int, Text, slice]
+                 ) -> Union["Edge", List["Edge"]]:
+    if isinstance(key, slice):
```

Comment:
Isn't this redundant with the `@overload`?

Reply:
`@overload`-decorated function signatures are only read by the type checker, and must be followed by a non-overload decorated function signature, at least according to the typing documentation.

Really I wanted to write 
```
  @overload
  def __getitem__(self, key: slice) -> List["Edge"]:
    pass 

  @overload
  def __getitem__(self, key: Union[int, Text]) -> "Edge":
    pass
```
but pytype didn't like this, initially. 

It seems like pytype is fine with the above now that I've demanded travis use a more updated version of pytype, so I've merged that.